src/Controller/GenericEntityController.ts

```diff
@@ -163,16 +163,18 @@ export class GenericEntityController<T extends BaseEntity> {
       };
     }
 
-    const isPublic = this.entity?.[this.formConfig.publicKey as keyof T] as boolean;
-
     this.entity = isUpdate ?
       await this.service.update(entityUpdateObject as T) :
       await this.service.add(entityUpdateObject as T);
 
-    if (isPublic) {
-      await this.service.setPublic(this.entity.id!);
-    } else {
-      await this.service.revokePublic(this.entity.id!);
+    if (this.formConfig?.publicKey) {
+      const isPublic = this.entity?.[this.formConfig.publicKey as keyof T] as boolean;
+
```

Comment:
Refactoring this to have the value of `this.formConfig?.publicKey` has a variable might enhance readability.

Reply:
Fixed in ebd3276be2d113c43782886a0d4ca665f8d575b4.